apps/mobile/src/hooks/useNotificationPayload.test.ts

```diff
@@ -0,0 +1,57 @@
+import { renderHook } from '@/src/tests/test-utils'
+import { useNotificationPayload } from './useNotificationPayload'
+import { useAuthGetNonceV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/auth'
+import { useSiwe } from '@/src/hooks/useSiwe'
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+
+import { Wallet } from 'ethers'
+
+jest.mock('@safe-global/store/gateway/AUTO_GENERATED/auth')
+jest.mock('@/src/hooks/useSiwe')
+jest.mock('@/src/store/hooks/activeSafe')
+jest.mock('@/src/store/hooks', () => ({
+  useAppSelector: (selector: (state: unknown) => unknown) => {
+    if (selector.name === 'selectSafeInfo') {
+      return { SafeInfo: { owners: [{ value: 'owner1' }] } }
+    }
+    return {}
+  },
+}))
+jest.mock('@/src/store/safesSlice')
+jest.mock('@/src/utils/logger')
+
+describe('useNotificationPayload', () => {
+  const mockCreateSiweMessage = jest.fn()
+  const mockUseAuthGetNonceV1Query = useAuthGetNonceV1Query as jest.Mock
+  const mockUseSiwe = useSiwe as jest.Mock
+  const mockUseDefinedActiveSafe = useDefinedActiveSafe as jest.Mock
+
+  beforeEach(() => {
+    mockUseAuthGetNonceV1Query.mockReturnValue({ data: 'mockNonce' })
```

Comment:
fyi: you should be able to use msw and just mock a return object for the nonce endpoint. 

by defining a handler in config/test/msw/handlers.ts this mocked data would be available in any test you do and you won't have to explicitly mock @safe-global/store/gateway/AUTO_GENERATED/auth